lib/replace-in-file.js

```diff
@@ -60,33 +61,49 @@ module.exports = function replaceInFile(config, cb) {
   }
 
   //Initialize helper vars
-  var totalFiles = config.files.length;
+  var totalFiles = 0;
   var processedFiles = 0;
   var changedFiles = [];
 
   //No callback given? Perform sync operation
   if (!cb) {
     config.files.forEach(function(file) {
-      if (replaceSync(file, config.replace, config.with)) {
-        changedFiles.push(file);
-      }
+      //Expand globs
+      glob.sync(file, {nodir: true}).forEach(function(file) {
+        if (replaceSync(file, config.replace, config.with)) {
+          changedFiles.push(file);
+        }
+      });
     });
     return changedFiles;
   }
 
   //Otherwise, perform async operation
   config.files.forEach(function(file) {
-    replace(file, config.replace, config.with, function(error, hasChanged) {
-      if (error) {
-        return cb(error);
-      }
-      if (hasChanged) {
-        changedFiles.push(file);
+    //Expand globs
+    glob(file, {nodir: true}, function(err, files) {
+      if (err) {
+        return cb(err);
       }
-      processedFiles++;
-      if (processedFiles === totalFiles) {
-        cb(null, changedFiles);
+      //Glob doesn't error if no files match, so we do:
+      if (files.length === 0) {
+        return cb(new Error('No files match the pattern: ' + file));
       }
+      files.forEach(function(file) {
+        totalFiles++;
```

Comment:
Since you have moved the totalFiles increment in the same `forEach` loop as the check on line 102, I don't think this will work for multiple files, as it's possible that totalFiles and processedFiles will both be 1 at the same time. Or do you rely on the fact that the replace operation will be executed slightly later than the processing of all files?

Maybe we should determine totalFiles after line 91 by simply stating `totalFiles = files.length`, before the `forEach` loop starts.